weaviate/collection/data.py

```diff
@@ -320,13 +327,35 @@ def __parse_properties_grpc(data: Dict[str, Any]) -> weaviate_pb2.BatchObject.Pr
                             uuids=val.uuids_str, prop_name=key
                         )
                     )
+            elif isinstance(val, list) and isinstance(val[0], bool):
+                bool_arrays.append(weaviate_pb2.BooleanArrayProperties(key=key, vals=val))
+            elif isinstance(val, list) and isinstance(val[0], str):
+                text_arrays.append(weaviate_pb2.TextArrayProperties(key=key, vals=val))
+            elif isinstance(val, list) and isinstance(val[0], datetime.datetime):
+                text_arrays.append(
+                    weaviate_pb2.TextArrayProperties(
+                        key=key, vals=[self.__datetime_to_string(x) for x in val]
+                    )
+                )
```

Comment:
Will the gRPC handler here be changed in the future to specific `DateArrayProperties` or is there no different between a date and a string across the network?

Reply:
there is no difference, weavite figures it out based on the datatype in the schema